src/login/logind-session.c

```diff
@@ -1237,6 +1268,12 @@ bool session_may_gc(Session *s, bool drop_not_started) {
         if (!s->user)
                 return true;
 
+        r = pidref_is_alive(&s->leader);
+        if (r < 0)
+                log_debug_errno(r, "Unable to determine if leader PID " PID_FMT " is still alive, assuming not.", s->leader.pid);
```

Comment:
... and thus the log message incorrect, since we might get `-ESRCH` as a valid result.

Reply:
I'm not sure if `pidref_is_alive` return is handled correctly. It will return -ESRCH if the field is unset. Maybe `pidref_is_alive` should be changed to return false instead?

Reply:
I think it is OK to put session in closing state if we can't figure out whether the leader is alive and I think it doesn't really matter what the reason is.